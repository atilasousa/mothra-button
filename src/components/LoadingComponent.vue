<template>
    <div class="loader" :class="getSize()"></div>
</template>
<script setup lang="ts">

const props = withDefaults(defineProps<{
    size?: "md" | "sm" | "lg";
}>(), {
    size: "md",

})

function getSize(): string {
    return `loader-${props.size}`
}
</script>
<style scoped lang="scss">
$background: #ffffff50;
$circle: #ffffff;

.loader {
    border: 4px solid $background;
    border-radius: 50%;
    border-top: 4px solid $circle;
    width: 12px;
    height: 12px;
    -webkit-animation: spin 2s linear infinite;
    animation: spin 1s linear infinite;
}

.loader-sm {
    border: 3px solid $background;
    border-top: 3px solid $circle;
    width: 10px;
    height: 10px;
}

.loader-lg {
    border: 5px solid $background;
    border-top: 5px solid $circle;
    width: 15px;
    height: 15px;
}

@-webkit-keyframes spin {
    0% {
        -webkit-transform: rotate(0deg);
    }

    100% {
        -webkit-transform: rotate(360deg);
    }
}

@keyframes spin {
    0% {
        transform: rotate(0deg);
    }

    100% {
        transform: rotate(360deg);
    }
}
</style>