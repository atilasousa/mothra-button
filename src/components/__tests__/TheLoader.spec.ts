import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import LoadingComponent from '../LoadingComponent.vue'

describe('LoadingComponent', () => {
    it('Checking if component is rendered', () => {
        const wrapper = mount(LoadingComponent)
        expect(wrapper.html()).toMatchSnapshot()
    })
})

describe('BaseButton / Checking Component Size', () => {
    it('Check if LoadingComponent medium size is render', () => {
      const wrapper = mount(LoadingComponent, { props: { size: 'md' } })
      expect(wrapper.classes()).toContain('loader-md')
    })
    it('Check if LoadingComponent Small size is render', () => {
      const wrapper = mount(LoadingComponent, { props: { size: 'sm' } })
      expect(wrapper.classes()).toContain('loader-sm')
    })
    it('Check if LoadingComponent Large size is render', () => {
      const wrapper = mount(LoadingComponent, { props: { size: 'lg' } })
      expect(wrapper.classes()).toContain('loader-lg')
    })
  })