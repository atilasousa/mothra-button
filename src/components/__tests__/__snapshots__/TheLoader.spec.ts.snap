// Vitest Snapshot v1

exports[`LoadingComponent > Checking if component is rendered 1`] = `"<div class=\\"loader loader-md\\" data-v-9770a80b=\\"\\"></div>"`;

exports[`LoadingComponent > should render correctly 1`] = `"<div class=\\"loader loader-md\\" data-v-9770a80b=\\"\\"></div>"`;
